src/fairchem/core/models/base.py

```diff
@@ -232,77 +233,86 @@ def forward(self, data: Batch) -> dict[str, torch.Tensor]:
-                self.backbone,
-                **head_config,
+        # This is required for hydras with models that have multiple outputs per head, since we will deprecate
+        # the old config system at some point, this will prevent the need to make major modifications to the trainer
+        # because they all expect the name of the outputs directly instead of the head_name.property_name
+        self.pass_through_head_outputs = pass_through_head_outputs
+
+        # if finetune_config is provided, then attempt to load the model from the given finetune checkpoint
+        starting_model = None
+        if finetune_config is not None:
+            starting_model: HydraModel = load_model_and_weights_from_checkpoint(finetune_config["starting_checkpoint"])
+            assert isinstance(starting_model, HydraModel), "Can only finetune starting from other hydra models!"
+
+        if backbone is not None:
+            backbone = copy.deepcopy(backbone)
+            backbone_model_name = backbone.pop("model")
+            self.backbone: BackboneInterface = registry.get_model_class(
+                backbone_model_name
+            )(
+                **backbone,
             )
+        elif starting_model is not None:
+            self.backbone = starting_model.backbone
+        else:
+            raise RuntimeError("Backbone not specified and not found in the starting checkpoint")
+
+        if heads is not None:
+            heads = copy.deepcopy(heads)
+            # Iterate through outputs_cfg and create heads
+            self.output_heads: dict[str, HeadInterface] = {}
+
+            head_names_sorted = sorted(heads.keys())
+            assert len(set(head_names_sorted)) == len(head_names_sorted), "Head names must be unique!"
+            for head_name in head_names_sorted:
+                head_config = heads[head_name]
+                if "module" not in head_config:
+                    raise ValueError(
+                        f"{head_name} head does not specify module to use for the head"
+                    )
 
-        self.output_heads = torch.nn.ModuleDict(self.output_heads)
+                module_name = head_config.pop("module")
+                self.output_heads[head_name] = registry.get_model_class(module_name)(
+                    self.backbone,
+                    **head_config,
+                )
 
-    def to(self, *args, **kwargs):
-        if "device" in kwargs:
-            self.device = kwargs["device"]
-        return super().to(*args, **kwargs)
+            self.output_heads = torch.nn.ModuleDict(self.output_heads)
+        elif starting_model is not None:
+            self.output_heads = starting_model.output_heads
+        else:
+            raise RuntimeError("Heads not specified and not found in the starting checkpoint")
 
     def forward(self, data: Batch):
+        # get device from input, at least one input must be a tensor to figure out it's device
```

Comment:
This looks good to me. 

nit; Is it possible to do just device=data.pos.device ? That might be a bit safer, since it wont have to iterate over all values on each batch. 

My only worry here is performance , but i dont think it should be too bad if the length of data.values() is reasonable ... and the runtime of a single batch is >500ms ?

Reply:
i think this line should be negligible, i'd rather not assume anything about whats in the data batch, but i can just save the device and only do this once as well